server/src/features/users/user.controller.js

```diff
@@ -128,4 +130,195 @@ export default class UserController {
       next(err);
     }
   }
+
+  /**
+   * Toggle user block status
+   * PATCH /api/users/:userId/block-status
+   * Body: { action: 'block' | 'unblock' }
+   */
+  static async toggleBlockStatus(req, res, next) {
+    try {
+      const { userId } = req.params;
+      const { action } = req.body;
+
+      // Validate request
+      const { error } = toggleBlockUserSchema.validate({ userId, action });
+      if (error) {
+        return res.status(400).json({
+          success: false,
+          message: error.details[0].message,
+        });
+      }
+
+      // Call service to handle the block/unblock logic
+      const result = await UserService.toggleBlockStatus(
+        req.user._id, // current admin's ID
+        userId,       // target user's ID
+        action        // 'block' or 'unblock'
+      );
+
+      return res.status(200).json({
+        success: true,
+        message: result.message,
+        data: {
+          userId: result.userId,
+          status: result.status,
+        },
+      });
+    } catch (err) {
+      // Pass the error to the error handling middleware
+      next(err);
+    }
+  }
+
+  /**
+   * Add an event to user's favorites
+   * POST /api/users/favorites
+   * Body: { eventId: string }
+   */
+  static async addToFavorites(req, res, next) {
+    try {
+      const { eventId } = req.body;
+      const userId = req.user._id;
+
+      // Validate request
+      const { error } = favoriteEventSchema.validate({ eventId });
+      if (error) {
+        return res.status(400).json({
+          success: false,
+          message: error.details[0].message,
+        });
+      }
+
+      // Check if user is allowed to add to favorites
+      const allowedRoles = ['student', 'staff', 'ta', 'professor'];
+      if (!allowedRoles.includes(req.user.role)) {
+        return res.status(403).json({
+          success: false,
+          message: 'Only students, staff, TAs, and professors can add events to favorites',
+        });
+      }
+
+      // Check if event exists
+      const event = await mongoose.model('Event').findById(eventId);
+      if (!event) {
+        return res.status(404).json({
+          success: false,
+          message: 'Event not found',
+        });
+      }
+
+      // Add to favorites if not already added
+      const user = await User.findById(userId);
```

Comment:
```suggestion
const user = await User.findByIdAndUpdate(
  userId,
  { $addToSet: { favoriteEvents: eventId } }, // atomic add
  { new: true, fields: { favoriteEvents: 1 } } // dont return the full user object
);

```
Why: atomic operation, prevents duplicates even under concurrency and Instead of returning the whole user object, return only fields you

Reply:
done